packages/core/src/metrics/platform/CordovaAppMetrics.ts

```diff
@@ -17,16 +19,29 @@ export class CordovaAppMetrics implements Metrics {
    * @returns {Promise<AppMetrics>} A promise containing the app metrics
    */
   public collect(): Promise<AppMetrics> {
-    const app = cordova.getAppVersion;
-
-    return Promise.all([
-      app.getPackageName(),
-      app.getVersionNumber()
-    ])
-      .then(results => ({
-        appId: results[0],
-        appVersion: results[1],
-        sdkVersion: version
-      }));
+    return new Promise((resolve, reject) => {
+      if (!document) {
+        return Promise.reject("Metrics not running in browser environment");
+      }
+      document.addEventListener("deviceready", () => {
```

Comment:
maybe refactor the event subscription into something like `platformUtils.whenDeviceReady().then(() => {...})`